eip_operator/src/controller/node.rs

```diff
@@ -64,6 +66,44 @@ impl k8s_controller::Context for Context {
         if matched_eips.is_empty() {
             return Err(Error::NoEipResourceWithThatNodeSelector);
         }
+
+        // Check the existing Node for EIP association and ready status.
+        // Node's that go in to a "NotReady" or "Unknown" state should have their EIP
+        // disassociated to allow a new node to spawn and use the EIP.
+        let node_condition_ready_status = node
+            .status
+            .as_ref()
+            .and_then(|status| status.conditions.as_ref())
+            .and_then(|conditions| conditions.iter().find(|c| c.type_ == "Ready"))
+            .map(|condition| condition.status.clone())
+            .ok_or(Error::MissingNodeReadyCondition)?;
+        match node_condition_ready_status.as_str() {
+            // Skip Ready Status True and continue with EIP node association.
+            "True" => {}
+            // Detach the EIP from a node with an Unknown or False ready status.
+            // An Unknown ready status could mean the node is unresponsive or experienced a hardware failure.
+            // A NotReady ready status could mean the node is experiencing a network issue.
+            _ => {
+                // Skip detachment if no EIP is associated with this node.
+                let node_eip = matched_eips.iter().find(|eip| {
+                    eip.status.as_ref().is_some_and(|s| {
+                        s.resource_id.is_some()
+                            && s.resource_id.as_ref().map(|r| r == &name).unwrap_or(false)
+                    })
+                });
+                if let Some(eip) = node_eip {
+                    warn!(
+                        "Node {} is in an unresponsive state, detaching EIP {}",
```

Comment:
Along with any requested changes, i'll adjust this to better reflect how the code is only doing changing the CRD - not the AWS EIP.

`"Node {} is in an unresponsive state, setting status detached on EIP {}",`